### PR TITLE
Remove v1alpha1 and v1alpha2 apis/controllers

### PR DESCRIPTION
**Pull request by pugangxa**
Delete v1alpha1 and v1alpha2 apis/controllers

### CODE REVIEW

**Review by terrytangyuan — COMMENTED** [review]
Any context for this change?

**Comment by pugangxa** [comment]
> Any context for this change?

@terrytangyuan Not sure whether it's needed, we are experimenting mpi-operator and based on the comments https://github.com/kubeflow/mpi-operator/pull/434#issuecomment-963233738 and also I saw there's issue https://github.com/kubeflow/mpi-operator/issues/422
Let me know if it's not needed,  thanks.

**Comment by alculquicondor** [comment]
The objective is to update the SDK containing v1 and v2 APIs, but the tools don't work without the matching dependencies.

And from the meeting some time ago, we mentioned that it was fine to remove v1alpha1 and v1alpha2 code. We could still leave the APIs, though, if you think they are necessary.

**Comment by alculquicondor** [comment]
@pugangxa for future reference, always link to context/evidence in the PR description so that it's easy for people to review and it's easy for people to find the reasons for a change in the future.

**Comment by terrytangyuan** [comment]
Ok that’s fine to me.

**Comment by pugangxa** [comment]
@alculquicondor @terrytangyuan Thanks for your comments. Please take a look so that I can continue generating V2 SDK task.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ We use Go v1.15+ for development and use [Go Modules](https://blog.golang.org/us
 ## Controller versions
 
 The main module `github.com/kubeflow/mpi-operator` contains the code of the legacy
-controllers `v1alpha1`, `v1alpha2` and `v1`.
+controllers `v1`.
```
Comment (alculquicondor) [inline CONTRIBUTING.md:22]: controller

File: go.mod [hunk 2]
```diff
@@ -4,27 +4,28 @@ go 1.13
 
 require (
```
Comment (alculquicondor) [inline go.mod:5]: Could you leave the upgrades to a separate PR?

File: CONTRIBUTING.md (hunk 1 above)
Comment (pugangxa) [reply CONTRIBUTING.md:22]: Thanks, will update.

File: go.mod (hunk 2 above)
Comment (pugangxa) [reply go.mod:5]: Sure, will do since that will be more concise.

**Comment by pugangxa** [comment]
This PR is just for remove v1alpha1 and v1alpha2 apis/controllers now, the bump version changes will be submitted in a separate PR.

**Review by terrytangyuan — APPROVED** [review]
Thanks!

/lgtm
/approve
